lib/timeutils/wallclocktime.c

```diff
@@ -873,6 +873,47 @@ wall_clock_time_guess_missing_year(WallClockTime *self)
     }
 }
 
+void
+wall_clock_time_guess_missing_fields(WallClockTime *self)
+{
+  /*
+   * The missing cases for date can be divided into three types:
+   * 1) missing all fileds -> use current date
+   * 2) only miss year -> guess year based on current year and month (current
+   * year, last year or next year)
+   * 3) the rest of the cases don't make much sense, so zero initialization of
+   * the missing field makes sense. And the year is initializeed to the current
+   *  one.
+   */
+  time_t now;
+  struct tm tm;
+
+  now = cached_g_current_time_sec();
+  cached_localtime(&now, &tm);
+
+  if(self->wct_year == -1 && self->wct_mon == -1 && self->wct_mday == -1)
+    {
+      self->wct_year = tm.tm_year;
+      self->wct_mon = tm.tm_mon;
+      self->wct_mday = tm.tm_mday;
+    }
+  else if(self->wct_year == -1 && self->wct_mon != -1 && self->wct_mday != -1)
+    {
+      self->wct_year = determine_year_for_month(self->wct_mon, &tm);
+    }
+  else
+    {
+      self->wct_year = (self->wct_year == -1)? tm.tm_year : self->wct_year;
+      self->wct_mon = (self->wct_mon == -1)? 0 : self->wct_mon;
+      self->wct_mday = (self->wct_mday == -1)? 1 : self->wct_mday;  // day of the month - [1, 31]
+    }
+
+  self->wct_hour = (self->wct_hour == -1)? 0 : self->wct_hour;
```

Comment:
[optional change]
Please do not use this style of setting the variable to itself when it does not require a change.
```c
bar = (bar == -1) ? 12 : bar;
```
The version when you only set to if needed is preferred:
```c
if (bar == -1)
   bar = 12;
```

The preference is based on readability and code style generally used in the project.